### ISSUE TITLE
Docker Compose Required

### ISSUE THREAD

**Issue by zlmitchell** [microsoft/vscode-remote-release#6263]
I tried the insiders build without success

- VSCode Version: 1.63.2
- Local OS Version: Windows 10 21H1 19043.1466
- Remote OS Version: 0.209.6
- Remote Extension/Connection Type: Docker
- Logs:
```
[86 ms] Remote-Containers 0.209.6 in VS Code 1.63.2 (899d46d82c4c95423fb7e10e68eba52050e30ba3).
[85 ms] Start: Resolving Remote
[95 ms] Setting up container for folder or workspace: c:\Users\zlmitche\gitlab\su-wagtail
[98 ms] Start: Check Docker is running
[98 ms] Start: Run: docker version
[845 ms] Server API version: Client:
 Version:           20.10.9
 API version:       1.41
 Go version:        go1.16.8
 Git commit:        c2ea9bc
 Built:             Thu Nov 18 21:19:50 2021
 OS/Arch:           windows/amd64
 Context:           default
 Experimental:      true

Server:
 Engine:
  Version:          20.10.11
  API version:      1.41 (minimum version 1.12)
  Go version:       go1.17.3
  Git commit:       847da184ad5048b27f5bdf9d53d070f731b43180
  Built:            Fri Nov 19 03:06:35 2021
  OS/Arch:          linux/amd64
  Experimental:     false
 containerd:
  Version:          v1.5.9
  GitCommit:        1407cab509ff0d96baa4f0eb6ff9980270e6e620
 runc:
  Version:          1.0.2
  GitCommit:        52b36a2dd837e8462de8e01458bf02cf9eea47dd
 docker-init:
  Version:          0.19.0
  GitCommit:
[960 ms] Start: Run: docker compose version --short
[990 ms] findLocalWindowsExecutable: Exectuable 'docker compose' not found on PATH 'C:\Python39\Scripts\;C:\Python39\;C:\Program Files (x86)\Common Files\Oracle\Java\javapath;C:\windows\system32;C:\windows;C:\windows\System32\Wbem;C:\windows\System32\WindowsPowerShell\v1.0\;C:\windows\System32\OpenSSH\;C:\ProgramData\chocolatey\bin;C:\Program Files\Git\cmd;C:\Program Files\Microsoft VS Code\bin;C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common;C:\WINDOWS\system32;C:\WINDOWS;C:\WINDOWS\System32\Wbem;C:\WINDOWS\System32\WindowsPowerShell\v1.0\;C:\WINDOWS\System32\OpenSSH\;C:\Program Files\nodejs\;C:\Program Files\PuTTY\;C:\Program Files\PowerShell\7\;C:\Program Files\Microsoft VS Code Insiders\bin;C:\Users\zlmitche\AppData\Local\Microsoft\WindowsApps;C:\Users\zlmitche\AppData\Local\GitHubDesktop\bin;C:\Users\zlmitche\AppData\Local\Microsoft\WindowsApps;C:\Program Files (x86)\Nmap;C:\Users\zlmitche\AppData\Roaming\npm;C:\Users\zlmitche\AppData\Local\Programs\Rancher Desktop\resources\resources\win32\bin;C:\Users\zlmitche\AppData\Local\Programs\Rancher Desktop\resources\resources\linux\bin'.
[993 ms] Start: Run: docker compose version --short
[1854 ms] Start: Run: docker ps -q -a --filter label=com.docker.compose.project=su-wagtail --filter label=com.docker.compose.service=webapp
[2601 ms] Start: Run: docker compose -f c:\Users\zlmitche\gitlab\su-wagtail\docker-compose.yml -f c:\Users\zlmitche\gitlab\su-wagtail\.devcontainer\docker-compose.yml --profile * config
[3437 ms] Command failed: docker compose -f c:\Users\zlmitche\gitlab\su-wagtail\docker-compose.yml -f c:\Users\zlmitche\gitlab\su-wagtail\.devcontainer\docker-compose.yml --profile * config
```
Steps to Reproduce:

1. Install Rancher Desktop as a Docker replacement (we are a large company with only a couple Developers)
2. Make sure that Docker and "Docker Compose" work
3. Update the Remote - Containers extension to use "docker compose" vs "docker-compose"
4. Try to open in remote container
If you do not update the "docker-compose" you will receive "Docker Compose Required" message which has about 0 logging. If you change the command to "docker compose" then you get an error when the container tries to start.

Leaving docker-compose i receive the following error
```
[1085 ms] Start: Run: docker-compose version --short
[1177 ms] Start: Run: docker ps -q -a --filter label=com.docker.compose.project=su-wagtail --filter label=com.docker.compose.service=webapp
[2066 ms] Command failed: docker-compose version --short
```

Does this issue occur when you try this locally?: Yes
Does this issue occur when you try this locally and all extensions are disabled?: Yes

**Comment by chrmarti** [comment]
It should automatically use `docker compose` when it can't find `docker-compose`. The setting for the Docker Compose path can only take an executable name or path, but no arguments.

Could you retry with VS Code 1.64 and Remote-Containers 0.217 and append the log again? Use the default setting:
<img width="517" alt="image" src="https://user-images.githubusercontent.com/9205389/152501085-89ddd198-b227-4212-96a1-26eaa44fd600.png">

**Comment by zlmitchell** [comment]
Still does not work, i get an alert dialog "Docker Compose required". Meanwhile in the PowerShell terminal in vscode i can type docker-compose just fine
```
PS > Get-Command docker-compose

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     docker-compose.exe                                 0.0.0.0    C:\Python39\Scripts\docker-compose.exe
```
remoteContainers log
```
[2022-02-08T21:04:35.144Z] Start: Run: docker version --format {{.Server.APIVersion}}
[2022-02-08T21:04:35.144Z] Remote-Containers 0.217.1 in VS Code 1.64.0 (5554b12acf27056905806867f251c859323ff7e9).
[2022-02-08T21:04:35.846Z] Stop (702 ms): Run: docker version --format {{.Server.APIVersion}}
[2022-02-08T21:04:35.847Z] 1.41
[2022-02-08T21:04:35.962Z] Start: Run: docker-compose version --short
[2022-02-08T21:04:36.022Z] Stop (60 ms): Run: docker-compose version --short
[2022-02-08T21:04:45.501Z] Start: Run: docker version --format {{.Server.APIVersion}}
[2022-02-08T21:04:46.227Z] Stop (726 ms): Run: docker version --format {{.Server.APIVersion}}
[2022-02-08T21:04:46.228Z] 1.41
```

**Comment by chrmarti** [comment]
Try using "C:\Python39\Scripts\docker-compose.exe" for the above setting.

**Comment by zlmitchell** [comment]
Setting the value as the default 'docker-compose', 'docker compose' or 'C:\Python39\Scripts\docker-compose.exe' have the same results with the error 'Docker Compose Required.'
'docker-compose' and 'docker compose' work from the PowerShell or command prompt terminal

Here is the output of my PATH from PowerShell
```
PS > $($Env:PATH).Split(';') 
C:\Python39\Scripts\
C:\Python39\        
C:\Program Files (x86)\Common Files\Oracle\Java\javapath
C:\windows\system32
C:\windows
C:\windows\System32\Wbem
C:\windows\System32\WindowsPowerShell\v1.0\
C:\windows\System32\OpenSSH\
C:\ProgramData\chocolatey\bin
C:\Program Files\Git\cmd
C:\Program Files\Microsoft VS Code\bin
C:\Program Files (x86)\NVIDIA Corporation\PhysX\Common
C:\WINDOWS\system32
C:\WINDOWS
C:\WINDOWS\System32\Wbem
C:\WINDOWS\System32\WindowsPowerShell\v1.0\
C:\WINDOWS\System32\OpenSSH\
C:\Program Files\nodejs\
C:\Program Files\PuTTY\
C:\Program Files\PowerShell\7\
C:\Program Files\Microsoft VS Code Insiders\bin
C:\Users\zlmitche\AppData\Local\Microsoft\WindowsApps
C:\Users\zlmitche\AppData\Local\GitHubDesktop\bin
C:\Users\zlmitche\AppData\Local\Microsoft\WindowsApps
C:\Program Files (x86)\Nmap
C:\Users\zlmitche\AppData\Roaming\npm
C:\Python39\Scripts\

C:\Users\zlmitche\AppData\Local\Programs\Rancher Desktop\resources\resources\win32\bin
C:\Users\zlmitche\AppData\Local\Programs\Rancher Desktop\resources\resources\linux\bin
```
Here is Get-Command docker*
```
PS > get-command docker*

CommandType     Name                                               Version    Source
-----------     ----                                               -------    ------
Application     docker.exe                                         20.10.9.0  C:\Users\zlmitche\AppData\Local\Programs\Rancher Desktop\resources\resources\win32\bin\docker.exe
Application     docker-compose.exe                                 0.0.0.0    C:\Python39\Scripts\docker-compose.exe
```

**Comment by chrmarti** [comment]
What do you get when you run `C:\Python39\Scripts\docker-compose.exe version --short` in a terminal?

Not familiar with Rancher Desktop, does that not include Docker Compose? Is `C:\Python39\Scripts\docker-compose.exe` a manual installation of Docker Compose?

**Comment by chrmarti** [comment]
Does `docker compose version --short` (no dash) work? If it does, try removing docker-compose.exe from the PATH (or uninstall it) and retry. (`docker compose` is version 2 of Docker Compose. Maybe update Rancher Desktop if it is not there yet.)

**Comment by zlmitchell** [comment]
```
PS> docker compose version --short
v2.2.3
```

**Comment by zlmitchell** [comment]
After ```pip3 uninstall docker-compose``` this appears to have finally worked. NOTE: that the remote containers settings is still targeting ```docker-compose```. Thank you @chrmarti I think this was resolved in the 1.64/0.217 change. I still feel that the docker-compose in python should be able to work as it does everywhere else... not sure what is up with that. ```docker compose``` and ```docker-compose``` do have some syntax differences.

**Comment by chrmarti** [comment]
Not sure why that wouldn't work, but using Docker Compose bundled with Rancher Desktop is the right approach. Thanks.
